### PR TITLE
Adds RSS feed for Camper News

### LINKED ISSUE #1597: Adding RSS feed for Camper News
I love the links I find on Camper News, and while I like both FCC and my own Camper News Zipline, I'd love to have the links aggregated along with my other RSS feeds to get more control and avoid having to repeatedly visit FCC/my Zipline.

I tried looking for a way to integrate it into my own Zipline but it seems that this can only be done on the back-end.

To add an RSS feed to the site, I'd need to add a feed.xml.jade template that looks *something* like this

```
doctype xml
rss(version="2.0", xmlns:atom="http://www.w3.org/2005/Atom")
  channel  
    title= title
    link= url
    description= description
    atom:link(href="#{ url }/feed.xml", rel="self", type="application/rss+xml")
    for post, slug in posts
      item
        title= post.title
        description <![CDATA[!{ partial(slug) }]]>
        pubDate= new Date(post.date)
        link #{ url }/news/#{ slug }
        guid(isPermaLink="true") #{ url }/news/#{ slug }
```

And then *something* like this in `<head>` of the page that advertises the feed. Can either be included in universal header or in a specific page. Since Camper News is just a part of FCC, I think just adding it to the stories page would suffice.

```
link(rel="alternate" type="application/rss+xml" title="{{ site.name }}" href="{{ site.url }}/feed.xml")
```

That's all I can figure out so far, I'd like some pointers in the right direction.

Thanks!

### PR DESCRIPTION
**Pull request by bugron**
Looks like this: http://prntscr.com/90d3bu.
Let me know if I should move `feed.jade` to somewhere else, add/remove/change something, thanks.
Closes #1597.

### CODE REVIEW

**Comment by SaintPeter** [comment]
Please add a newline at the end of `feed.jade`.

Otherwise, looks good.

File: server/boot/story.js [hunk 1]
```diff
@@ -64,6 +64,7 @@ module.exports = function(app) {
 
   router.post('/news/userstories', userStories);
   router.get('/news/hot', hotJSON);
+  router.get('/feed', RSSFeed);
   router.get('/stories/hotStories', hotJSON);
```
Comment (BerkeleyTrue) [inline server/boot/story.js]: let's move this to `/news/feed`

Comment (bugron) [inline server/boot/story.js]: Done.

**Comment by BerkeleyTrue** [comment]
Looks like we need to filter out links that are not valid. Also I guess the date is not in the correct format.
feed_validator_results__http___beta_freecodecamp_com_news_feed

**Comment by bugron** [comment]
@BerkeleyTrue I'll see what I can do. What validator do you use?

**Comment by BerkeleyTrue** [comment]
https://validator.w3.org/feed/ but I had to throw this PR on beta to test it.

**Comment by bugron** [comment]
1.Date format is fixed.
2, 3. 4.Not sure how is that possible to have a camper news post with a link `http://` because FCC already filters links (I'm getting `Please enter a valid URL format (http://www.example.com/).` warning on the submit page). Also links in camper news are unique that is why I've used them as GUIDs.
5.Feed is defined as an `XML` document so not sure what is the issue. See [this link](https://validator.w3.org/feed/docs/warning/UnexpectedContentType.html).
6.The link is hardcoded so in fcc.com this won't be an issue.
7.Will take a deeper look.

**Comment by BerkeleyTrue** [comment]
Sounds good.

**Comment by bugron** [comment]
Hopefully the issue with HTML entities encoding is solved, please check if I need to `replace` any other entity too.

**Comment by SaintPeter** [comment]
I get the following errors:
image

### Steps to test locally

1. Go to http://www.freecodecamp.com/news/hot
2. Save output locally as `story.json`
3. Import the json into your local instance with this command:  
`mongoimport --db freecodecamp --collection story --jsonArray --file story.json`
4. Go to http://localhost:3001/news/feed
5. Save as a raw .HTML file
6. Open file in a text editor
7. Copy and paste into https://validator.w3.org/feed/#validate_by_input

**Comment by bugron** [comment]
@BerkeleyTrue For FCC camper news link validator `http:///` is a valid link!

**Comment by BerkeleyTrue** [comment]
@bugron I've deleted the offending stories.

**Comment by bugron** [comment]
@BerkeleyTrue awesome. This should filter such links.

**Comment by SaintPeter** [comment]
:sparkles: :+1: :sparkles:

**Pull request merged**